differ/storage.go

```diff
@@ -523,22 +534,27 @@ func (storage DBStorage) WriteNotificationRecordForCluster(
 		notifiedAt, errorLog)
 }
 
-// ReadLastNNotifiedRecords method returns the last N notification records
-// with state = 'sent' for given org ID and cluster name.
-func (storage DBStorage) ReadLastNNotifiedRecords(clusterEntry types.ClusterEntry,
-	numberOfRecords int) ([]types.NotificationRecord, error) {
-	var notificationRecords = make([]types.NotificationRecord, 0)
+// ReadLastNotifiedRecordForClusterList method returns the last notification
+// with state = 'sent' for given org IDs and clusters.
+func (storage DBStorage) ReadLastNotifiedRecordForClusterList(clusterEntries []types.ClusterEntry, timeOffset string) (types.NotifiedRecordsPerCluster, error) {
+	var orgIDs = make([]string, len(clusterEntries))
+	var clusterIDs = make([]string, len(clusterEntries))
+
+	for idx, entry := range clusterEntries {
+		orgIDs[idx] = strconv.FormatUint(uint64(entry.OrgID), 10)
+		clusterIDs[idx] = string(entry.ClusterName)
+	}
+
+	whereClause := fmt.Sprintf(` WHERE org_id IN (%v) AND cluster IN (%v) AND state = 1 `,
+		inClauseFromSlice(orgIDs), inClauseFromSlice(clusterIDs))
```

Comment:
does it even work for strings?

Reply:
WDYM for strings?

I'm thinking of simplifying the function to only expect []string, so the caller function would be responsible for passing the correct format. Something like

```
func inClauseFromStringSlice(slice []string) string {
    if len(slice) == 0 {
        return ""
    }
    return "'" + strings.Join(slice, `','`) + `'`
}
```